src/platform/ESP32/ConnectivityManagerImpl_WiFi.cpp

```diff
@@ -780,7 +780,7 @@ void ConnectivityManagerImpl::ChangeWiFiStationState(WiFiStationState newState)
         ChipLogProgress(DeviceLayer, "WiFi station state change: %s -> %s", WiFiStationStateToStr(mWiFiStationState),
                         WiFiStationStateToStr(newState));
         mWiFiStationState = newState;
-        SystemLayer().ScheduleLambda([]() { NetworkCommissioning::ESPWiFiDriver::GetInstance().OnNetworkStatusChange(); });
+        NetworkCommissioning::ESPWiFiDriver::GetInstance().OnNetworkStatusChange();
```

Comment:
Is this code already running on the Matter event queue or with the Matter lock otherwise held?  If not, is it ok to call OnNetworkStatusChange() without the Matter lock held?

Reply:
The scheduled function `OnNetworkingStatusChange` and the function in which the lambda is scheduled i.e `ChangeWifiStation` state run in the CHIP context i,e with task name as 'chip' irrespective of the lambda.

Reply:
Great, thanks!